filters.go

```diff
@@ -0,0 +1,107 @@
+package main
+
+import (
+	"regexp"
+
+	"fmt"
+
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func dataSourceFiltersSchema() *schema.Schema {
+	return &schema.Schema{
+		Type:     schema.TypeSet,
+		Optional: true,
+		ForceNew: true,
+		Elem: &schema.Resource{
+			Schema: map[string]*schema.Schema{
+				"name": {
+					Type:     schema.TypeString,
+					Required: true,
+				},
+
+				"values": {
+					Type:     schema.TypeList,
+					Required: true,
+					Elem:     &schema.Schema{Type: schema.TypeString},
+				},
+
+				"regex": {
+					Type:     schema.TypeBool,
+					Optional: true,
+					Default:  false,
+				},
+			},
+		},
+	}
+}
+
+// Process an entity's properties (string or array of strings) by N filter sets of
+// keyword:values, where each filter set ANDs and each keyword:values set ORs
+func ApplyFilters(filters *schema.Set, items []map[string]interface{}) []map[string]interface{} {
+	if filters == nil || filters.Len() == 0 {
+		return items
+	}
+
+	for _, f := range filters.List() {
+		fSet := f.(map[string]interface{})
+		keyword := fSet["name"].(string)
+
+		isReg := false
+		if regex, regexOk := fSet["regex"]; regexOk && regex != nil {
+			if strVal, strValOk := regex.(string); strValOk {
+				isReg = strVal == "1" || strVal == "true"
```

Comment:
At this point you could use https://golang.org/pkg/strconv/#ParseBool, which would also accept values such as TRUE.